src/attr/setters.pyi

```diff
@@ -0,0 +1,12 @@
+from . import _OnSetAttrType, Attribute
+from typing import TypeVar, Any, NewType, cast
+
+_T = TypeVar("_T")
+
+def pipe(*setters: _OnSetAttrType) -> _OnSetAttrType: ...
+def frozen(instance: Any, attribute: Attribute, new_value: Any) -> Any: ...
```

Comment:
I wonder if this could be:
```
from typing import NoReturn

def frozen(instance: Any, attribute: Attribute[_T], new_value: _T) -> NoReturn: ...
```

Reply:
NoReturn is quite fresh in typing, isn't it? I remember pulling it from mypy_extensions or something? 🤔

Reply:
Since this is a stub the only thing that matters is where it is in `typeshed`.

Reply:
Oh!